src/Microsoft.Cci.Extensions/Writers/CSharp/CSDeclarationWriter.cs

```diff
@@ -18,6 +18,7 @@ public partial class CSDeclarationWriter : ICciDeclarationWriter
         private bool _forCompilation;
         private bool _forCompilationIncludeGlobalprefix;
         private bool _forCompilationThrowPlatformNotSupported;
```

Comment:
Yeah combine them.  Let string.empty indicate no message and null indicate _forCompilationThrowPlatformNotSupported=false.  You don't have to force everyone to add a message, though you could...  Seems like there are some cases where we might want the default message?